developer/src/kmc-keyboard-info/test/test-keyboard-info-compiler.ts

```diff
@@ -176,6 +261,135 @@ describe('keyboard-info-compiler', function () {
+    }
+    const kpsFilename = KHMER_ANGKOR_KPS;
+    const kmpCompiler = new KmpCompiler();
+    assert.isTrue(await kmpCompiler.init(callbacks, {}));
+    const kmpJsonData = kmpCompiler.transformKpsToKmpObject(kpsFilename);
+    assert.isNotNull(kmpJsonData);
+    const origKmpCompilerTransformKpsToKmpObject = KmpCompiler.prototype.transformKpsToKmpObject;
+    kmpJsonData.files = testCase.files;
+    let result: KeyboardInfoCompilerResult;
+    try {
+      KmpCompiler.prototype.transformKpsToKmpObject = (_kpsFilename: string): KmpJsonFile.KmpJsonFile => kmpJsonData;
+      result = await compiler.run(kpjFilename, null);
+    } catch(e) {
+      assert.fail(e);
+    } finally {
+      KmpCompiler.prototype.transformKpsToKmpObject = origKmpCompilerTransformKpsToKmpObject;
+    }
+    assert.isNotNull(result);
+    const keyboard_info = JSON.parse(new TextDecoder().decode(result.artifacts.keyboard_info.data));
+    assert.deepEqual(keyboard_info.packageIncludes.sort(), testCase.packageIncludes.sort());
+  }));
+
+  const minKeymanVersionTestCases = [
+    { js:  '4.0', omitJs:  true, kmx:  '4.0', expected:  '5.0' },
+    { js:  '4.0', omitJs: false, kmx:  '4.0', expected:  '5.0' },
+    { js: '10.0', omitJs: false, kmx:  '4.0', expected: '10.0' },
+    { js:  '4.0', omitJs: false, kmx: '10.0', expected: '10.0' },
+    { js: '10.0', omitJs: false, kmx: '10.0', expected: '10.0' },
+  ];
+
+  minKeymanVersionTestCases.forEach((testCase, idx) => it(`check run sets minKeymanVersion correctly (test case #${idx})`, async function() {
+    const kpjFilename = KHMER_ANGKOR_KPJ;
+    const sources = KHMER_ANGKOR_SOURCES;
+    const compiler = new KeyboardInfoCompiler();
+    assert.isTrue(await compiler.init(callbacks, {sources}));
+    const origCompilerLoadJsFile = compiler['loadJsFile'];
+    const origKmxFileVersionToString = compiler['kmxFileVersionToString'];
+    let jsFile = compiler['loadJsFile'](sources.jsFilename);
+    const insert = testCase.omitJs ? '' : `this.KMINVER="${testCase.js}";`;
+    jsFile = jsFile.replace('this.KMINVER="10.0";', insert);
+    compiler['loadJsFile'] = (_filename: string) => jsFile;
+    compiler['kmxFileVersionToString'] = (_version: number) => testCase.kmx;
+    const result = await compiler.run(kpjFilename, null);
+    compiler['loadJsFile'] = origCompilerLoadJsFile;
+    compiler['kmxFileVersionToString'] = origKmxFileVersionToString;
+    assert.isNotNull(result);
+    const keyboard_info = JSON.parse(new TextDecoder().decode(result.artifacts.keyboard_info.data));
+    assert.deepEqual(keyboard_info.minKeymanVersion, testCase.expected);
+  }));
+
+  const platformsTestCases = [
+    { hasJsFile: true, targets: 'any',                  expected: { windows: "full",macos: "full",linux: "full",desktopWeb: "full",ios: "full",android: "full",mobileWeb: "full" } },
+    { hasJsFile: false, targets: '',                    expected: { desktopWeb: "full",mobileWeb: "full" } },
```

Comment:
This doesn't seem right -- if there is no .js file, then it can't target web platforms?

Reply:
The whole snippet shows:

https://github.com/keymanapp/keyman/blob/c1b7f0053d762a3b5cd8e0e3c710992290994224/developer/src/kmc-keyboard-info/src/keyboard-info-compiler.ts#L327-L337

So, that has `jsFile == true`?

Reply:
Hmm ... more precision needed in my reply! (line numbers from beta)

The test is focused on the setting of the platforms supported if there is a `.js` file that has been supplied to the compiler via the options.sources in `init()` (`keyboard-info-compiler.ts` ln 125, 151) and successfully loaded from disk (ln 179).  The test explores several possibilities, including no platform metadata in the `.kmx` files (`keyboard-info-compiler.ts` ln 328-337), and whether the presence of various platforms implies mobile or desktop web too (ln 346-351).

The test uses the Khmer Angkor default fixtures, but stubs in two places to force the code down alternate paths over nine test cases.  The two stubs are to remove the `.js` file from the `.kps` file list produced by the KmpCompiler (`keyboard-info-compiler.ts` ln 162) stubbed in `test-keyboard-info-compiler.ts` ln 365-368, 379 and to set the targets from the `.kmx` file (`keyboard-info-compiler.ts` ln 185-188) stubbed in `test-keyboard-info-compiler.ts` ln 369-374, 380.

The choice of element tag `hasJsFile` is misleading, as all test cases have a `.js` file, so `hasJsFileInKps` would be better.  I will change this in #11255.